src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscribeTerminator.cs

```diff
@@ -39,9 +39,10 @@ public MessageDrivenSubscribeTerminator(SubscriptionRouter subscriptionRouter, s
                 subscriptionMessage.Headers[Headers.ReplyToAddress] = subscriberAddress;
                 subscriptionMessage.Headers[Headers.SubscriberTransportAddress] = subscriberAddress;
                 subscriptionMessage.Headers[Headers.SubscriberEndpoint] = subscriberEndpoint.ToString();
-                var address = publisherAddress;
+                subscriptionMessage.Headers[Headers.TimeSent] = DateTimeExtensions.ToWireFormattedString(DateTime.UtcNow);
```

Comment:
I wonder if it would make sense to have some kind of extension method which sets the standard headers.